### PR TITLE
Chatrooms

### PR DESCRIPTION
**Pull request by JordanVlieg**
Hey guys, sorry this one may be tough to review.  I did a lot including linting and bugfixing.  Please make sure you are using 4 spaces instead of tabs in your dev env.  

The big change here is that I implemented chat rooms and they work.  Please take a look and let me know what you think.

### CODE REVIEW

File: theapp/chatapp.py [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
+def on_join(json):
+    alias = json['alias']
+    room = json['room']
+    join_room(room)
+    emit('join response', json, room=room)
+    #emit(alias + ' has entered the room.', room=room)
+
+@socketio.on('leave')
+def on_leave(json):
+    username = json['username']
```
Comment (SethTolkamp) [inline theapp/chatapp.py]: Should username be alias?

File: theapp/static/js/chat.js [hunk 2]
```diff
@@ -34,38 +35,32 @@ var chosen_color = getRandomColor();
...
-// Capture Message
-socket.on('chat', function( msg ){
-	// FIX THIS msg.user
-	if( typeof msg.user !== 'undefined' ){
-		$('h1').remove();
-		$('div.message_holder').append('<div class="message_roll"><b style="color:' + msg.color + '">' + msg.user + ': </b>' + msg.msg + '</div>' );
-		$('#chatbox').scrollTop($('#chatbox')[0].scrollHeight);
-		$('#chatbox').css('max-height', $(window).height() - 150);
-		// clear textbox
-		$('#chatinputbox').val('');
```
Comment (SethTolkamp) [inline theapp/static/js/chat.js:69]: We still need this line dont we?

**Review by imandrewli — APPROVED** [review]
I will find another way to clear the input field on message sent.

File: theapp/static/js/chat.js (hunk 2 above)
Comment (JordanVlieg) [reply theapp/static/js/chat.js:69]: No, the functionality has been moved.  This is the code that caused bugs in our demo.

File: theapp/chatapp.py (hunk 1 above)
Comment (JordanVlieg) [reply theapp/chatapp.py]: Leave is not implemented.  Ill delete it.

**Pull request merged**